src/main/java/hudson/remoting/PingThreadFileListCap.java

```diff
@@ -0,0 +1,83 @@
+package hudson.remoting;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Maintains most recent N files in a directory in cooperation with the writer.
+ *
+ * Useful to record incidents as files without bankrupting
+ *
+ */
+public class PingThreadFileListCap {
```

Comment:
does this need to be public - seems its internal and there is IIRC something in Jenkins to perform this for other users.